gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/deploy/StoreDeployer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2015-16 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.config.store.deploy;
+
+import gobblin.config.store.api.ConfigStore;
+import gobblin.config.store.api.ConfigStoreFactory;
+
+import java.net.URI;
+import java.util.ServiceLoader;
+
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * A tool to deploy configs in the classpath to a {@link ConfigStore}. It finds the config files in classpath under the
+ * <code>classpathStoreRoot</code> directory. Every config file found will be deployed to the <code>storeUri</code> with
+ * a new <code>version</code>. The deployment semantics are defined the {@link ConfigStore} themselves. A
+ * {@link ConfigStore} must implement {@link Deployable} for the {@link StoreDeployer} to deploy on it. If the
+ * {@link ConfigStore} for <code>storeUri</code> does not implement {@link Deployable}, the deployment will be a no-op
+ */
+@Slf4j
+public class StoreDeployer {
+
+  /**
+   * Deploy configs in <code>classpathStoreRoot</code> to <code>storeUri</code>
+   *
+   * @param storeUri to which confgs are deployed
+   * @param classpathStoreRoot the top level directory in classpath for all configs to be deployed
+   * @param version to be used for this deployment
+   *
+   */
+  @SuppressWarnings({ "rawtypes", "unchecked" })
+  public static void deploy(URI storeUri, String classpathStoreRoot, String version) throws Exception {
+
+    ServiceLoader<ConfigStoreFactory> loader = ServiceLoader.load(ConfigStoreFactory.class);
+
+    for (ConfigStoreFactory storeFactory : loader) {
+
+      log.info("Found ConfigStore with scheme : " + storeFactory.getScheme());
+
+      if (storeUri.getScheme().equals(storeFactory.getScheme())) {
+
+        log.info("Using ConfigStore with scheme : " + storeFactory.getScheme());
+
+        ConfigStore configStore = storeFactory.createConfigStore(storeUri);
+
+        if (configStore instanceof Deployable<?, ?>) {
+
+          ((Deployable) configStore).deploy(new FsDeploymentConfig(classpathStoreRoot, version));
```

Comment:
This makes this class tied to `FsDeploymentConfig` what if a different config needs to be used?